File: leetcode/easy/check-if-matrix-is-x-matrix.ts
// https://leetcode.com/problems/check-if-matrix-is-x-matrix/

// MINE
function checkXMatrix(grid: number[][]): boolean {
  const n = grid.length
  for (let i = 0; i < n; ++i) {
    for (let j = 0; j < n; ++j) {
      if (i === j || i === n - j - 1) {
        if (!grid[i][j]) return false
      } else {
        if (grid[i][j]) return false
      }
    }
  }
  return true
}

// ANSWER

File: leetcode/easy/check-if-matrix-is-x-matrix.py
# https://leetcode.com/problems/check-if-matrix-is-x-matrix/

class Solution:
    def checkXMatrix(self, grid: List[List[int]]) -> bool:
        for i in range(len(grid)):
            for j in range(len(grid)):
                if i == j or i+j == len(grid)-1:
                    if grid[i][j] == 0:
                        return False
                else:
                    if grid[i][j] > 0:
                        return False
        return True
